server/server.go

```diff
@@ -114,21 +118,16 @@ func (s *server) Shutdown() {
 
 func (s *server) newErrorEncoder() kithttp.ErrorEncoder {
 	return func(ctx context.Context, err error, w http.ResponseWriter) {
-		switch e := err.(type) {
-		case kithttp.Error:
-			err = e.Err
-
-			switch e.Domain {
-			case kithttp.DomainEncode:
-				w.WriteHeader(http.StatusBadRequest)
-			case kithttp.DomainDecode:
-				w.WriteHeader(http.StatusBadRequest)
-			case kithttp.DomainDo:
-				w.WriteHeader(http.StatusBadRequest)
-			default:
-				w.WriteHeader(http.StatusInternalServerError)
-			}
-		default:
+		rErr := err.(microserver.ResponseError)
+		uErr := rErr.Underlying()
+
+		if rErr.IsEndpoint() {
+			json.NewEncoder(w).Encode(map[string]interface{}{
+				"code":    microserver.CodeFailure,
+				"message": uErr.Error(),
+				"from":    s.serviceName,
+			})
```

Comment:
I was wondering why this is but I saw that this was also done this way in kubernetesd. This is in fact the default structure microkit already uses. See https://github.com/giantswarm/microkit/blob/1ac51fb9905e0c7ab5a8d12fec127c6fbfd75aac/server/server.go#L392-L394. So it is not necessary to do this. We probably implemented this during some transition period and forgot to clean it up. It should just be like this: https://github.com/giantswarm/api/blob/master/server/server.go#L236. You can set the error code and error message to the response error. This makes the code easier. Lets also make sure we clean this up in kubernetesd. You can go for it or I will do it later. However you like.